### PR TITLE
refs sparkfabrik-innovation-team/board#2957: remove cache and add artifacts

### PR DESCRIPTION
**Pull request by FabrizioCafolla**

### CODE REVIEW

File: templates/gitlab-ci-template-cloud-run.yml [hunk 1]
```diff
@@ -164,9 +146,16 @@ test variables:
 build:
   stage: build
   extends: .build_template
+  artifacts:
+    expire_in: 1 day
```
Comment (paolomainardi) [inline templates/gitlab-ci-template-cloud-run.yml:150]: @Syphon83 just wondering if we can keep the value lower than this ? how the garbage collection is managed on the gitlab side ?

Comment (Syphon83) [reply templates/gitlab-ci-template-cloud-run.yml:150]: Artifacts smallest granularity is seconds. 
`artifacts:
    expire_in: 30 seconds`

If no expiring date has been declared there's an instance wide value in the admin area. 

By default the most recent artifact is always kept if the pipeline succeded.

File: templates/gitlab-ci-template-cloud-run.yml [hunk 2]
```diff
@@ -164,9 +146,16 @@ test variables:
...
+    paths:
+      - ./docker-images
 
 # The following jobs run only for the branch defined in the `DEPLOYMENT_MAIN_BRANCH` variable.
 deploy:
+  needs:
+    - job: build
+      artifacts: true
```
Comment (Monska85) [inline templates/gitlab-ci-template-cloud-run.yml:158]: The `needs` keyword is not needed, the jobs are in two different stages.

File: templates/gitlab-ci-template-cloud-run.yml [hunk 3]
```diff
@@ -164,9 +146,16 @@ test variables:
 build:
   stage: build
   extends: .build_template
+  artifacts:
```
Comment (Monska85) [inline templates/gitlab-ci-template-cloud-run.yml:149]: I would prefer to define the `artifacts` definition in the template and not in the final job.

**Review by Monska85 — CHANGES_REQUESTED** [review]

File: templates/gitlab-ci-template-cloud-run.yml (hunk 1 above)
Comment (paolomainardi) [reply templates/gitlab-ci-template-cloud-run.yml:150]: we can set a lower value, i think that `1 hour` is more than enough

**Review by paolomainardi — APPROVED** [review]
LGTM

**Pull request merged**